Enterprise-CMCS/managed-care-review
# MCR-4147: submitContract updated for contract API

### Summary

This PR
- Updates the submitContract resolver to not rely on and call the submitHealthPlanPackage resolver (removing the submitHPP resolver and test helper functions will happen in a [subsequent ticket](https://jiraent.cms.gov/browse/MCR-4552))
- Updates the CMS and State emails for submit and resubmit to use contract API
- Creates a zod parser for the contract

#### Related issues

https://jiraent.cms.gov/browse/MCR-4529

#### Test cases covered

- submitContract resolver: ported over and updated existing test
- zod parser function: happy path and checked expected errors for invalid data

## QA guidance

- Manual Q&A - check submit flows for submissions with and without linked rates. I also checked that unlock and resubmit were working
- Include unhappy path in QAing for this work. Can use the graphql explorer update endpoint to get a submission in a bad state (check the parseContract and submitContract for logic that will trigger errors)
- On Hana's suggestion I tested this with val, someone else might want to try it with prod data to confirm before merging


Specific Review of the Code
- Double check the logic of the zod parser (parseContract) that lives in `dataValidatorHelpers.ts`
- Double check that the emails still look good (4 emails for submit and resubmit for CMS and state users), snapshots should all have been updated
